src/main/java/com/bookingApp/bookingApp/service/ReservationService.java

```diff
@@ -0,0 +1,181 @@
+            s.setTitle(screening.getMovie().getTitle());
+            s.setRoom(screening.getSeatScreenings().get(0).getSeat().getRoom().getRoomNumber());
+
+            screeningsDtos.add(s);
+        });
+
+        Collections.sort(screeningsDtos, new Comparator<ScreeningsDto>() {
+            @Override
+            public int compare(ScreeningsDto o1, ScreeningsDto o2) {
+                if(o1.getTitle().compareTo(o2.getTitle())==0){
+                    return o1.getDate().compareTo(o2.getDate());
+                }else {
+                    return o1.getTitle().compareTo(o2.getTitle());
+                }
+            }
+        });
+
+        return screeningsDtos;
+    }
+    @Override
+    public List<ScreeningsDto> findAll(LocalDateTime dateFrom, LocalDateTime dateTo) {
+        List<ScreeningsDto> screeningsDtos = new ArrayList<>();
+        List<Screening> screenings = screeningRepository.readScreeningsByScreeningDateBetween(dateFrom,dateTo);
+        screenings.forEach(screening -> {
+            ScreeningsDto s = new ScreeningsDto();
+            s.setScreeningId(screening.getId());
+            s.setDate(screening.getScreeningDate());
+            s.setTitle(screening.getMovie().getTitle());
+            s.setRoom(screening.getSeatScreenings().get(0).getSeat().getRoom().getRoomNumber());
+
+            screeningsDtos.add(s);
+        });
+
+        Collections.sort(screeningsDtos, new Comparator<ScreeningsDto>() {
+            @Override
+            public int compare(ScreeningsDto o1, ScreeningsDto o2) {
+                if(o1.getTitle().compareTo(o2.getTitle())==0){
+                    return o1.getDate().compareTo(o2.getDate());
+                }else {
+                    return o1.getTitle().compareTo(o2.getTitle());
+                }
+            }
+        });
+        return screeningsDtos;
+    }
+
+    @Override
+    public ScreeningInfoDto findById(Long id) {
+        ScreeningInfoDto s = new ScreeningInfoDto();
+        Screening screening = screeningRepository.findById(id)
+                .orElseThrow(()->new ScreeningNotFoundException(id));
+        List<SeatDto> seatDtos = new ArrayList<>();
+
+        s.setScreeningId(screening.getId());
+        s.setDate(screening.getScreeningDate());
+        s.setTitle(screening.getMovie().getTitle());
+        s.setRoom(screening.getSeatScreenings().get(0).getSeat().getRoom().getRoomNumber());
+        for (SeatScreening seatScreening: screening.getSeatScreenings()) {
+            SeatDto seatDto = new SeatDto();
+            seatDto.setSeatId(seatScreening.getSeat().getId());
+            seatDto.setSeatNumber(seatScreening.getSeat().getSeatNumber());
+            seatDto.setReserved(seatScreening.isReserved());
+
+            seatDtos.add(seatDto);
+        }
+        s.setSeats(seatDtos);
+
+        return s;
+    }
+
+    @Override
+    @Transactional
+    public ReservationBackDto reserve(ReservationDto reservation, Long screeningId){
```

Comment:
czy ta metoda nie robi za dużo?

Reply:
faktycznie moglbym podzielic to na dwie metody, jedna ktora wystawia bilety i druga ktora obsluguje rezerwacje